rcl/test/rcl/test_timer.cpp

```diff
@@ -12,6 +12,13 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// Suppress deprecated function warning
+#ifndef _WIN32
+# pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#else  // defined(_WIN32)
+# pragma warning(disable: 4996)
+#endif
```

Comment:
I added this declarations in order to avoid the warnings of deprecation during the CI (which were causing it to 'fail'). On the test, I added it for the whole file because there are several places where `rcl_timer_init` was used and would reduce readability to add it one by one. Let me know if you think this is ok or if we want to add the `# pragma GCC diagnostic ignored "-Wdeprecated-declarations"` only to the places where `rcl_timer_init` is being called.

Reply:
So the thing is, we don't want to do either.  If we are going to deprecate an API, then all of the in-core uses of it should use the new API, not the old one.

So the correct fix here is to go through and fix all of the callers of the old API to the new one.  Unfortunately, this also means that we'll probably have to change `rclcpp` and `rclpy` as well.

Reply:
Oh ok, I misunderstood what we wanted to do I'll do the modifications. Regarding the other packages, yes it affects `rclcpp` package (I already made the changes there on this [PR](https://github.com/ros2/rclcpp/pull/2005) and regarding `rclpy` I'll do the changes and comment that PR here.

Reply:
This are the repositories with their corresponding PR that were modified due to this change:
- rclcpp -> [#2005](https://github.com/ros2/rclcpp/pull/2005)
- rclpy -> [#999](https://github.com/ros2/rclpy/pull/999)
- system_test -> [#508](https://github.com/ros2/system_tests/pull/508)